pkg/storage/db.go

```diff
@@ -65,115 +65,19 @@ const latestStatusQuery = `WITH latest_candidates AS (
 `
 
 // Query to find resource by private IP using v2 schema
-// nolint
-const resourceByPrivateIPQuery = `select ia.private_ip,
-       res.arn_id,
-       res.meta,
-       ar.region,
-       rt.resource_type,
-       aa.account,
-	   aa.id
-from aws_private_ip_assignment ia
-         left join aws_resource res on ia.aws_resource_id = res.id
-         left join aws_region ar on res.aws_region_id = ar.id
-         left join aws_resource_type rt on res.aws_resource_type_id = rt.id
-         left join aws_account aa on res.aws_account_id = aa.id
-where ia.private_ip = $1
-  and ia.not_before < $2
-  and (ia.not_after is null or ia.not_after > $2)
-`
+const resourceByPrivateIPQuery = `select * from get_resource_by_private_ip($1, $2)`
```

Comment:
This massively improves readability, and is a good alternative to the other solution we discussed (sqlc - https://github.com/kyleconroy/sqlc). I wonder if we can combine both (sqlc also gives us golang stucts per row for "free")...
NIT: Did you consider calling via https://www.postgresql.org/docs/11/sql-syntax-calling-funcs.html#SQL-SYNTAX-CALLING-FUNCS-NAMED , I think it would make the queries inside the code even more readable.

Reply:
Cool suggestion, i will look into incorporating them. Thank you!

Reply:
On further review of sqlc, i think sqlc is powerful tool. Sqlc will definitely help us simplify our code by reducing boiler plate code. As followup i think we could track SQLC specific work as part of separate task as i do see changes being significant atleast in db.go code. Thoughts?

Reply:
Of course, it's different scope. I mostly wondered how well would sqlc work with functions. Might want to spike this later.